crates/precompile/src/bls12_381/g1_msm.rs

```diff
@@ -34,7 +35,7 @@ pub(super) fn g1_msm(input: &Bytes, gas_limit: u64) -> PrecompileResult {
     }
 
     let mut g1_points: Vec<_> = Vec::with_capacity(k);
-    let mut scalars_bytes: Vec<u8> = Vec::with_capacity(k * SCALAR_LENGTH);
+    let mut scalars: Vec<_> = Vec::with_capacity(k);
```

Comment:
```suggestion
    let mut scalars = Vec::with_capacity(k);
```
We can probably remove  `Vec<_>` fully